apps/rush-lib/src/cli/utilities/pnpm/PnpmLinkManager.ts

```diff
@@ -0,0 +1,190 @@
+      RushConstants.rushTempProjectsFolderName, unscopedTempProjectName);
+
+    // Example: "C:\MyRepo\common\temp\projects\project1\package.json"
+    const packageJsonFilename: string = path.join(extractedFolder, 'package.json');
+
+    // Example: "C:\MyRepo\common\temp\node_modules\@rush-temp\project1"
+    const installFolderName: string = path.join(this._rushConfiguration.commonTempFolder,
+      RushConstants.nodeModulesFolderName, RushConstants.rushTempNpmScope, unscopedTempProjectName);
+
+    const commonPackage: BasePackage = BasePackage.createVirtualTempPackage(packageJsonFilename, installFolderName);
+
+    const localPackage: BasePackage = BasePackage.createLinkedPackage(
+      project.packageJson.name,
+      commonPackage.version,
+      project.projectFolder,
+      commonPackage
+    );
+
+    // now that we have the temp package.json, we can go ahead and link up all the direct dependencies
+
+    // first, start with the rush dependencies, we just need to link to the project folder
+    Object.keys(commonPackage.packageJson!.rushDependencies || {}).forEach((dependencyName: string) => {
+
+      const matchedRushPackage: RushConfigurationProject | undefined =
+        this._rushConfiguration.getProjectByName(dependencyName);
+
+      if (matchedRushPackage) {
+        const matchedVersion: string = matchedRushPackage.packageJson.version;
+
+        let localLinks: string[] = rushLinkJson.localLinks[localPackage.name];
+        if (!localLinks) {
+          localLinks = [];
+          rushLinkJson.localLinks[localPackage.name] = localLinks;
+        }
+        localLinks.push(dependencyName);
+
+        // We did not find a suitable match, so place a new local package that
+        // symlinks to the Rush project
+        const newLocalFolderPath: string = path.join(localPackage.folderPath, 'node_modules', dependencyName);
+
+        const newLocalPackage: BasePackage = BasePackage.createLinkedPackage(
+          dependencyName,
+          matchedVersion,
+          newLocalFolderPath
+        );
+
+        newLocalPackage.symlinkTargetFolderPath = matchedRushPackage.projectFolder;
+        localPackage.children.push(newLocalPackage);
+      } else {
+        // weird state or program bug
+        throw Error('Cannot find rush dependency in rush configuration');
+      }
+    });
+
+    // Iterate through all the regular dependencies
+
+    // With NPM, it's possible for two different projects to have dependencies on
+    // the same version of the same library, but end up with different implementations
+    // of that library, if the library is installed twice and with different secondary
+    // dependencies.The NpmLinkManager recursively links dependency folders to try to
+    // honor this. Since PNPM always uses the same physical folder to represent a given
+    // version of a library, we only need to link directly to that folder, and it will
```

Comment:
Probably not true.  The point of your change in this PR is to ensure that how/when PNPM handles this problem, Rush will copy whatever PNPM is doing